src/Bicep.Core/Parsing/BaseParser.cs

```diff
@@ -1037,13 +1039,13 @@ protected IEnumerable<Token> NewLines()
             }
         }
 
-        private IntegerLiteralSyntax NumericLiteral()
+        private (Token literal, ulong value) NumericLiteral()
```

Comment:
>private [](http://example.com/codeflow?start=8&length=7)

Do we also need to make updates to the grammar.md pseudo grammar file

Reply:
The grammar isn't changing, just the AST. So where previously we would have always used `IntegerLiteralSyntax`, we now use `IntegerLiteralSyntax` in a value expression and `IntegerTypeLiteralSyntax` in a type expression. That way there's no ambiguity later in the compilation process as to whether this syntax node was part of a type or a value.